### DISCUSSION TITLE
Parallel/Distributed computations using River?

### DISCUSSION THREAD

**Discussion by ZahirBilal** [online-ml/river#1021] [Q&A] [answered]
Hello River community,

I am conducting an evaluation for various River classifiers on a large dataset (almost 3 million data points). In the same experiment, I am also performing hyperparameter optimisation for the RandomSampling parameters, specifically fine-tuning the desired distribution as well as the sampling rate hyperparameters using some techniques.

I am running the experiment on a supercomputer. However, I noticed that the python file (the task) is only processed with one processor. My question: is there any way to distribute the computational load across all 30+ CPUs?

Thanks a lot
Zahir

**Comment by BaAL94** [comment]
Hey Zarhir,

yes it is possible I think, I didn't try it with river yet but with Skmultiflow it did work. therefore you need to install the multiprocess library:
https://multiprocess.readthedocs.io/en/latest/

**Comment by gbolmier** [comment]
Hi @ZahirBilal, Python [`multiprocessing`](https://docs.python.org/3/library/multiprocessing.html) module from the standard library can help you to parallelise independent experiment runs. Given that you are apparently using a large dataset you should also account for RAM consumption across your parallel runs.

**Comment by MaxHalford** [comment]
As said above, parallel workloads is a broad and general topic. You'll find loads of resources online on how to do this.

**Comment by ZahirBilal** [comment, marked as answer]
Thanks for your inputs guys. Unfortunately that did not work, I misclarified my question, that task that I need to parallelize is actually the hyperparameter optimization using hyperopt. Multiprocessing does not work with hyperopt, as hyperopt's optimization function is minizing the loss using a gradient descent algorithm, this need special treatment to parallelize (either using a spakr cluster or MongoDB). 

thanks and best wishes,
Zahir
